src/main/stgeorge/validator/RequestValidate.java

```diff
@@ -0,0 +1,102 @@
+package stgeorge.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class RequestValidate {
+    private Map<String, String> data;
+
+    public RequestValidate(Map<String, String> data) {
+        this.data = data;
+    }
+
+    public  List<RequestValidationResult> validateRequest()
+    {
+        List<RequestValidationResult> results = new ArrayList<RequestValidationResult>();
+        if(data.isEmpty())
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"Request is Empty");
+            results.add(result);
+            return  results;
+        }
+        if(data.get("firstname").length() <=1)
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"FirstName is in valid");
+            results.add(result);
+
+        }
+        if(data.get("lastname").length() <=1)
+        {
+            RequestValidationResult result = new RequestValidationResult(false, "LastName is in valid");
+            results.add(result);
+        }
+        if(data.get("emailaddress").length() <=4)
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"emailaddress is in valid");
+            result.setMessage("emailaddress is in valid");
+            results.add(result);
+        }
+        if(data.get("phonenumber").length() <10)
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"phonenumber is invalid");
+            results.add(result);
+        }
+        if(data.get("case")=="contactus")
```

Comment:
== will not work with String in Java. use .equals("contactus")

Also its good to use switch() for this?

Reply:
Done